pkg/processor/runtime/golang/test/golang_test.go

```diff
@@ -196,6 +202,95 @@ func (suite *TestSuite) TestCustomEvent() {
 	})
 }
 
+func (suite *TestSuite) TestFileStream() {
+	createFunctionOptions := suite.GetDeployOptions("file-streamer",
+		path.Join(suite.GetTestFunctionsDir(), "common", "file-streamer", "golang"))
+
+	tempDir := os.TempDir()
+	defer os.RemoveAll(tempDir)
+
+	createFunctionOptions.FunctionConfig.Spec.Volumes = []functionconfig.Volume{
+		{
+			Volume: v1.Volume{
+				Name: "tmp",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: tempDir,
+					},
+				},
+			},
+			VolumeMount: v1.VolumeMount{
+				Name:      "tmp",
+				MountPath: tempDir,
+			},
+		},
+	}
+
+	for _, testRequest := range []struct {
+		responseSize    int
+		repeat          int
+		requestBody     string
+		deleteAfterSend bool
+	}{
+		{
+			responseSize:    4 * 1024,
+			repeat:          20,
+			requestBody:     common.GenerateRandomString(256, common.LettersAndNumbers),
+			deleteAfterSend: false,
+		},
+		{
+			responseSize:    10 * 1024 * 1024,
+			deleteAfterSend: true,
+		},
+	} {
+		// generate a random file
+		tempFile, err := ioutil.TempFile(tempDir, "")
+		suite.Require().NoError(err)
+
+		responseBody := common.GenerateRandomString(testRequest.responseSize, common.LettersAndNumbers)
+
+		// write random stuff to the file which we'll expect as a response
+		_, err = tempFile.Write([]byte(responseBody))
+		suite.Require().NoError(err)
+
+		httpRequest := httpsuite.Request{
+			RequestBody:          testRequest.requestBody,
+			RequestMethod:        "POST",
+			RequestPath:          tempFile.Name(),
+			ExpectedResponseBody: responseBody,
+			ExpectedResponseHeaders: map[string]string{
+				"X-request-body": testRequest.requestBody,
+			},
+		}
+
+		if testRequest.deleteAfterSend {
+			httpRequest.RequestPath += "?delete_after_send=true"
+		}
+
+		suite.DeployFunction(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
+			httpRequest.Enrich(deployResult)
+
+			for repeatIdx := 0; repeatIdx < testRequest.repeat+1; repeatIdx++ {
+				if !suite.SendRequestVerifyResponse(&httpRequest) {
+					return false
+				}
+			}
+
+			return true
+		})
+
+		if !testRequest.deleteAfterSend {
+			err = os.Remove(tempFile.Name())
+			suite.Require().NoError(err)
+		} else {
+
+			// expect file to be removed
+			_, err = os.Stat(tempFile.Name())
+			suite.Require().Error(err)
+		}
```

Comment:
imo change condition to be positive so it would be easier to read (`not not` vs `true`)

```suggestion
		if testRequest.deleteAfterSend {
		        // expect file to be removed
			_, err = os.Stat(tempFile.Name())
			suite.Require().Error(err)
			
		} else {
			err = os.Remove(tempFile.Name())
			suite.Require().NoError(err)
		}
```